modules/@apostrophecms/modal/ui/apos/components/AposModal.vue

```diff
@@ -94,186 +94,196 @@
+  return props.modal.triggerFocusRefresh;
+});
+
+const hasBeenLocalized = computed(() => {
+  return Object.keys(apos.i18n.locales).length > 1;
+});
+
+const currentLocale = computed(() => {
+  return apos.i18n.locale;
+});
+
+function hasSlot(slotName) {
+  return Boolean(slots[slotName]);
+}
+
+const classes = computed(() => {
+  const classes = [ 'apos-modal' ];
+  classes.push(`apos-modal--${props.modal.type}`);
+  if (props.modal.type === 'slide') {
+    if (props.modal.origin) {
+      classes.push(`apos-modal--origin-${props.modal.origin}`);
+    } else {
+      classes.push('apos-modal--origin-right');
     }
+    classes.push('apos-modal--full-height');
+  }
+  if (props.modal.busy) {
+    classes.push('apos-modal--busy');
+  }
+  return classes.join(' ');
+});
+
+const innerClasses = computed(() => {
+  const classes = [];
+  if (props.modal.width) {
+    classes.push(`apos-modal__inner--${props.modal.width}`);
+  };
+
+  return classes;
+});
+
+const gridModifier = computed(() => {
+  const hasLeftRail = hasSlot('leftRail');
+  const hasRightRail = hasSlot('rightRail');
+
+  if (hasLeftRail && hasRightRail) {
+    return 'apos-modal__main--with-rails';
+  }
+  if (hasLeftRail && !hasRightRail) {
+    return 'apos-modal__main--with-left-rail';
+  }
+  if (!hasLeftRail && hasRightRail) {
+    return 'apos-modal__main--with-right-rail';
+  }
+  return false;
+});
+
+watch(triggerFocusRefresh, (newVal) => {
+  if (shouldTrapFocus.value) {
+    nextTick(trapFocus);
+  }
+});
+
+onMounted(() => {
+  if (shouldTrapFocus.value) {
+    nextTick(trapFocus);
+  }
+});
+
+function onKeydown(e) {
+  const hasPressedEsc = e.keyCode === 27;
+  if (hasPressedEsc) {
+    close(e);
+  }
+}
+
+function onEnter() {
+  emit('show-modal');
+  bindEventListeners();
+  apos.modal.stack = apos.modal.stack || [];
+
+  apos.modal.stack.push(state);
+  nextTick(() => {
+    emit('ready');
+  });
+}
+
+function onLeave() {
+  removeEventListeners();
+  emit('no-modal');
+
+  apos.modal.stack = apos.modal.stack
+    .filter(modal => modal.id !== state.id);
+
+  focusLastModalFocusedElement();
+}
+
+function bindEventListeners() {
+  window.addEventListener('keydown', onKeydown);
+}
+
+function removeEventListeners() {
+  window.removeEventListener('keydown', onKeydown);
+}
+
+function trapFocus() {
+  const elementSelectors = [
+    '[tabindex]',
+    '[href]',
+    'input',
+    'select',
+    'textarea',
+    'button'
+  ];
+
+  const selector = elementSelectors
+    .map(addExcludingAttributes)
+    .join(', ');
+
+  elementsToFocus.value = [ ...modalEl.value.querySelectorAll(selector) ]
+    .filter(isElementVisible);
+
+  focusElement(focusedElement.value, elementsToFocus.value[0]);
+
+  function addExcludingAttributes(element) {
+    return `${element}:not([tabindex="-1"]):not([disabled]):not([type="hidden"]):not([aria-hidden])`;
+  }
+}
+
+function close() {
+  if (apos.modal.stack.at(-1)?.id !== state.id) {
+    return;
   }
-};
+  emit('esc');
```

Comment:
Are the event listeners still active when I call close directly?

Reply:
Yes!